File: 771-JewelsandStones/771-JewelsandStones.java
/*

771. Jewels and Stones
Easy

You're given strings J representing the types of stones that are jewels, and S representing the stones you have.  Each character in S is a type of stone you have.  You want to know how many of the stones you have are also jewels.

The letters in J are guaranteed distinct, and all characters in J and S are letters. Letters are case sensitive, so "a" is considered a different type of stone from "A".

Example 1:

Input: J = "aA", S = "aAAbbbb"
Output: 3

Example 2:

Input: J = "z", S = "ZZ"
Output: 0

Note:

    S and J will consist of letters and have length at most 50.
    The characters in J are distinct.

*/

class Solution {
    public int numJewelsInStones(String J, String S) {
        int res = 0;
        Set<Character> set = new HashSet<>();
        for (int i =0; i<J.length(); i++)
            set.add(J.charAt(i));
        for (int i =0; i<S.length(); i++){
            if (set.contains(S.charAt(i)))
                res++;
        }
        return res;
    }
}

File: 771-JewelsandStones/771-JewelsandStones.py
"""

771. Jewels and Stones

You're given strings jewels representing the types of stones that are jewels, and stones representing the stones you have. Each character in stones is a type of stone you have. You want to know how many of the stones you have are also jewels.

Letters are case sensitive, so "a" is considered a different type of stone from "A".

Example 1:

Input: jewels = "aA", stones = "aAAbbbb"
Output: 3

Example 2:

Input: jewels = "z", stones = "ZZ"
Output: 0

Constraints:

    1 <= jewels.length, stones.length <= 50
    jewels and stones consist of only English letters.
    All the characters of jewels are unique.

"""

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jset = set()
        for ch in jewels: jset.add(ch)
        res = 0
        for ch in stones:
            if ch in jset: res += 1
        return res
